Indexing/LiteralSubstitutionTree.cpp

```diff
@@ -56,6 +56,14 @@ void LiteralSubstitutionTree::handleLiteral(Literal* lit, Clause* cls, bool inse
 
   Literal* normLit=Renaming::normalize(lit);
 
+  if(_theoryConstraints){
+    ASS(!lit->isEquality());
```

Comment:
We only use constraints with bianry resolution (amongst inferences that require literal unification). Binary resolution does not use equality.

Reply:
Okay fair. I think we should add a comment for that. Because I picture someone in the furture trying to use UWA for this, and thinking there might be any deep technical issue why things might break if you try to use the index with equalities.

I mean tbf it doesn't make much sense to use any literal index with equality and treating it like a normal literal because of it being commutative, buuuut anyways just a clarifying comment could help someone in the future debugging :)